kart/working_copy.py

```diff
@@ -240,6 +242,43 @@ def reset(
                 rewrite_full=rewrite_full,
             )
 
+    def set_state_post_commit(
```

Comment:
minor: I'd prefer `set_state_after_commit` because the word 'post' might mean some kind of action. However... is this the same as a 'soft reset' in git terminology? If so, what about we just call it that?

Reply:
Changed to `soft_reset_after_commit`.
(I don't love git's name for things - hard, soft and mixed resets are simply the name for those type of resets. What they mean simply to be learned since they don't, for instance, reset the hard/soft/mixed area of your working copy. Probably those resets should have been called things like `unstage`, `checkout --discard-changes`, `rebase-changes` or... I don't know. But I guess "set_state" also means nothing so soft_reset is at least step up from that. I've left in the "after_commit" since this is specifically the type of soft_reset we do after a commit - that's why we're clearing from the tracking table and remove features that don't match spatial filters. There's nothing in Git that's an exact correspondence)